corehq/apps/data_interfaces/tasks.py

```diff
@@ -93,7 +93,13 @@ def run_case_update_rules(now=None):
                .order_by('domain'))
     for domain in domains:
         if not any_migrations_in_progress(domain) and not DISABLE_CASE_UPDATE_RULE_SCHEDULED_TASK.enabled(domain):
-            run_case_update_rules_for_domain.delay(domain, now)
+            domain_obj = Domain.get_by_name(domain)
+            if domain_obj.auto_case_update_hour is None:
+                hour_to_run = settings.RULE_UPDATE_HOUR
+            else:
+                hour_to_run = domain_obj.auto_case_update_hour
+            if hour_to_run == datetime.utcnow().hour:
+                run_case_update_rules_for_domain.delay(domain, now)
```

Comment:
is it possible this takes longer than an hour to run (either due to the volume of rules or a backup in the periodic queue) and misses some tasks because the hour has rolled over? It might be safer to calculate `now` once at the beginning

Reply:
Maybe an alternative would be to save the last time this is run, and run all the tasks scheduled between then and the current hour, in case maintenance or something else causes an hour to get missed.

Reply:
I'm not especially worried about this task taking longer than an hour, because it should be quick to just kick off all the tasks, and even if it takes longer than an hour, won't the next task still get kicked off? But in any case, I added 4d5b58735bf to only call `utcnow` before the loop starts.

As far as maintenance goes, I'm leaning against doing special handling because it doesn't seem right to say HQ will do something at a particular time and then ultimately do it at a different time. If we have lengthy unexpected downtime, I'm not sure we want to immediately run everything that wasn't run during downtime once HQ is back up. Open to discussing this more, but for now I'm going to merge this PR.